### PR TITLE
FileUpload: Make component accessible by keyboard navigation

### LINKED ISSUE #46501: A11y : Cannot use FileUpload component via keyboard
**Steps to reproduce**:

[FileUpload component](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/FileUpload/FileUpload.tsx#L45-L56): Try to click "Upload JSON file" button inside import modal

**Actual Result**:

not accesible via keyboard

**Expected Result**

workflow accessible for keyboard-only users

**Relevant WCAG Criteria:**

[2.1.1 Keyboard - Level A](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=335%2C332#keyboard)

**Environment**:
- Grafana version: 8.4.3
- Data source type & version:
- User OS & Browser:
- Others:

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
This PR makes it possible to use file input component via keyboard.

**Which issue(s) this PR fixes**:

Fixes #46501

**Special notes for your reviewer**:

Additionally I have migrated from using the deprecated `stylesFactory` towards `useStyles2` hook

### CODE REVIEW

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx [hunk 1]
```diff
@@ -42,37 +43,48 @@ export const FileUpload: FC<Props> = ({
...
+const getStyles = (size: ComponentSize) => (theme: GrafanaTheme2) => {
   const buttonStyles = getButtonStyles({ theme, variant: 'primary', size, iconOnly: false });
+  const focusStyle = getFocusStyles(theme);
+
   return {
-    fileUpload: css`
-      display: none;
-    `,
-    button: buttonStyles.button,
+    fileUpload: css({
```
Comment (jackw) [inline packages/grafana-ui/src/components/FileUpload/FileUpload.tsx:74]: nit: can we align usage of the `css` function to use either the template literal or the object notation please?

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx [hunk 2]
```diff
@@ -42,37 +43,48 @@ export const FileUpload: FC<Props> = ({
...
     <>
-      <label className={cx(style.button, className)}>
+      <input
+        type="file"
+        id="fileUpload"
+        className={style.fileUpload}
+        onChange={onChange}
+        multiple={false}
+        accept={accept}
+        data-testid="fileUpload"
```
Comment (jackw) [inline packages/grafana-ui/src/components/FileUpload/FileUpload.tsx:53]: Is there any value in adding this to our e2e-selectors? 🤔

**Review by jackw — APPROVED** [review]
Tested in Chrome, Safari and Firefox and works as expected. Great work @tolzhabayev ! 🚀

**Comment by zoltanbedi** [comment]
We should also ask @JoaoSilvaGrafana as he also tried to tackle this in https://github.com/grafana/grafana/pull/46374

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx [hunk 3]
```diff
@@ -42,37 +43,48 @@ export const FileUpload: FC<Props> = ({
...
+      />
+      <label htmlFor="fileUpload" className={cx(style.labelWrapper, className)}>
```
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply packages/grafana-ui/src/components/FileUpload/FileUpload.tsx:55]: That's actually a good question - the `id`=`fileUpload` was already there before and should have been unique already in order to be used properly regardless of `label` wrapping - or am I missing something?

If I would change it to something auto-generated, wouldn't it be a breaking change for the ones relying on it?

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx (hunk 1 above)
Comment (tolzhabayev) [reply packages/grafana-ui/src/components/FileUpload/FileUpload.tsx:74]: https://github.com/grafana/grafana/pull/47497/commits/dda51b02f66d4733409b5c0542c5faab3a2970c4 done

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx [hunk 4]
```diff
@@ -42,37 +43,48 @@ export const FileUpload: FC<Props> = ({
...
-          className={style.fileUpload}
-          onChange={onChange}
-          multiple={false}
-          accept={accept}
-        />
       </label>
+
       {fileName && (
-        <span aria-label="File name" className={style.fileName}>
+        <span aria-label="File name" className={style.fileName} data-testid="fileName">
```
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply packages/grafana-ui/src/components/FileUpload/FileUpload.tsx]: Moved https://github.com/grafana/grafana/pull/47497/commits/ba06a8ef1f5f6c005a98a790a27e0c61919f22cb

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx (hunk 2 above)
Comment (tolzhabayev) [reply packages/grafana-ui/src/components/FileUpload/FileUpload.tsx]: Switched to using e2e-selectors

**Review by JoaoSilvaGrafana — APPROVED** [review]
Seems like a good solution so I closed my PR! Tested it and works 👍

File: packages/grafana-ui/src/components/FileUpload/FileUpload.test.tsx [hunk 5]
```diff
@@ -29,4 +31,19 @@ describe('FileUpload', () => {
     });
     expect(wrapper.find({ 'aria-label': 'File name' }).text()).toEqual('longFileName.png');
```
Comment (axelavargas) [inline packages/grafana-ui/src/components/FileUpload/FileUpload.test.tsx:32]: 🤔 should we remove/update the enzyme test above in this PR?

Comment (tolzhabayev) [reply packages/grafana-ui/src/components/FileUpload/FileUpload.test.tsx]: @axelavargas I rewrote the tests to use react testing library instead of enzyme

**Review by axelavargas — APPROVED** [review]
Fantastic job @tolzhabayev 👏🏾 , I think this PR also will close this issue https://github.com/grafana/grafana/issues/48292

File: packages/grafana-ui/src/components/FileUpload/FileUpload.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jackw) [reply packages/grafana-ui/src/components/FileUpload/FileUpload.tsx]: 👍

**Pull request merged**